src/selection/selection.js

```diff
@@ -15,6 +15,9 @@ import {
   SELECTION_TYPE_UNRECOGNIZED,
 } from './utils';
 import { toSingleLine } from './../helpers/templateLiteralTag';
+import { DefaultBorderStyle as CellDefaultBorderStyle } from './highlight/types/cell';
+import { DefaultBorderStyle as AreaDefaultBorderStyle } from './highlight/types/area';
+import { DefaultBorderStyle as FillDefaultBorderStyle } from './highlight/types/fill';
```

Comment:
I think that the Selection here knows too much about highlight types. BorderStyle management should be moved to the Highlight module. I've prepared a commit with a suggestion on how the whole idea can be improved. I hope that you'll agree on https://github.com/handsontable/handsontable/commit/4810c3414299b70691b4d1aa38daa5841a82fbdc

The changes touch several files including `selection.js`.